spec/magic_pipe/transports/https_spec.rb

```diff
@@ -119,5 +119,15 @@ def self.it_submits_a_request_with_the_correct_data
 
       it_submits_a_request_with_the_correct_data
     end
+
+    describe "when the HTTP request fails" do
+      before do
+        stub_request(:post, base_url).to_return(status: 504)
+      end
+
+      it "raises an exception" do
+        expect { perform }.to raise_error(MagicPipe::Transports::SubmitFailedError)
```

Comment:
it would be nice to see the message as well
```
expect { perform }.to raise_error(MagicPipe::Transports::SubmitFailedError, "#{transport_klass} couldn't submit message")
```